standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/common/ndv/hll/HLLSparseRegister.java

```diff
@@ -195,7 +199,7 @@ public void extractLowBitsTo(HLLRegister dest) {
       byte lr = entry.getValue(); // this can be a max of 65, never > 127
       if (lr != 0) {
         // should be a no-op for sparse
-        dest.add((long) ((1 << (p + lr - 1)) | idx));
+        dest.add((1 << (p + lr - 1)) | idx);
```

Comment:
nit: just noticing this. To be safe wrt overflow can you make it `1L << (p + lr - 1)`